activerecord/lib/active_record/connection_adapters/postgresql/oid/interval.rb

```diff
@@ -0,0 +1,42 @@
+require 'active_support/duration'
+
+module ActiveRecord
+  module ConnectionAdapters
+    module PostgreSQL
+      module OID # :nodoc:
+        class Interval < Type::Value # :nodoc:
+          def type
+            :interval
+          end
+
+          def type_cast_from_database(value)
+            if value.kind_of? ::String
+              ::ActiveSupport::Duration.parse(value)
+            else
+              super
+            end
+          end
+
+          def type_cast_from_user(value)
+            type_cast_from_database(value)
+          rescue ::ActiveSupport::Duration::ISO8601ParsingError
+            value # Allow user to supply raw string values in another formats supported by PostgreSQL
```

Comment:
Restricted. It will raise `ActiveSupport::Duration::ISO8601Parser::ParsingError` exception and will say what value is wrong:

    > Event.new(duration: '1 year 2 mons')
    ActiveSupport::Duration::ISO8601Parser::ParsingError: Invalid ISO 8601 duration: "1 year 2 mons"